lib/SensorFusion/src/SensorFusion.cpp

```diff
@@ -0,0 +1,112 @@
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and classes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Prototypes
+ *****************************************************************************/
+
+/******************************************************************************
+ * Local Variables
+ *****************************************************************************/
+
+/******************************************************************************
+ * Public Methods
+ *****************************************************************************/
+
+/******************************************************************************
+ * Protected Methods
+ *****************************************************************************/
+
+/******************************************************************************
+ * Private Methods
+ *****************************************************************************/
+
+void SensorFusion::init(void)
+{
+    m_lkf.init();
+}
+
+void SensorFusion::estimateNewState(SensorData newSensorData)
+{
+    /* Estimate the current angle. */
+    int16_t estimatedAngle;
+    estimateAngle(estimatedAngle, newSensorData.angleOdometry, newSensorData.magnetometerValueX, newSensorData.magnetometerValueY);
+
+    /* Transform the acceleration values from the robot coordinate system into the world coordinate system */
+    int16_t accelerationInRobotCoordinateSystem[2] = {newSensorData.accelerationX, newSensorData.accelerationY};
+    int16_t accelerationInGlobalCoordinateSystem[2];
+    transfromLocalToGlobal(accelerationInGlobalCoordinateSystem, accelerationInRobotCoordinateSystem, estimatedAngle);
+
+    /* perform the Kalman Filter Prediction and Update Steps */
+    m_lkf.predictionStep();
+    m_lkf.updateStep(newSensorData.accelerationX, newSensorData.accelerationY, newSensorData.positionOdometryX, newSensorData.positionOdometryY);
+}
+
+void SensorFusion::transfromLocalToGlobal(int16_t* globalResult, const int16_t* localVectorToTransform, const int16_t& rotationAngle)
+{
+    /*  Calculate the sin and cos of the rotationAngle; convert the angle from mrad to rad. */
+    float cosValue = cosf(static_cast<float>(rotationAngle) / 1000);
+    float sinValue = sinf(static_cast<float>(rotationAngle) / 1000);
+
+    globalResult[0] = cosValue * localVectorToTransform[0] - sinValue * localVectorToTransform[1];
+    globalResult[1] = sinValue * localVectorToTransform[0] + cosValue * localVectorToTransform[1];
+}
+
+void SensorFusion::estimateAngle(int16_t & estimatedAngle, const int16_t & encoderAngle, const int16_t & magnetometerValueX, const int16_t & magnetometerValueY)
```

Comment:
Please attach the reference modifier either to the type or to the name (to the type would be consistent with the rest of your code).